generic/piranha-tree-sitter/src/test.rs

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright (c) 2022 Uber Technologies, Inc.
+
+ <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ except in compliance with the License. You may obtain a copy of the License at
+ <p>http://www.apache.org/licenses/LICENSE-2.0
+
+ <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ express or implied. See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+use std::collections::HashMap;
+use std::fs::{self, DirEntry};
+use std::path::{Path, PathBuf};
+
+use colored::Colorize;
+use log::info;
+
+use crate::config::command_line_arguments::{CommandLineArguments, PiranhaArguments};
+use crate::piranha::{FlagCleaner, SourceCodeUnit};
+use crate::utilities::{initialize_logger, read_file};
+use std::sync::Once;
+
+static INIT: Once = Once::new();
```

Comment:
should this file go under the test directory? This and the `test-resources` can possibly move into a `test` directory which is a sibling to `src`?